lib/notifier.js

```diff
@@ -1,27 +1,68 @@
 const notifier = require('node-notifier');
 const { config } = require('./config');
 const telegramBot = require('./telegram-bot');
-const cache = {};
+const cache = {businessesByItem:{}};
 
 module.exports = {
     notifyIfChanged
 };
 
+function businessesToMap(businesses) {
+
+    return businesses.reduce(function(map, business) {
+        map[business.item.item_id] = business;
+        return map;
+    }, {});
+}
+
+function hasInterestingChange(current, last) {
+    const options = config.get('messageFilter');
+
+    const currentStock = current.items_available
+    const lastStock = last === undefined ? 0 : last.items_available
+    
+    if (currentStock == lastStock) {
+        return options.showUnchanged;
+    } else if (currentStock < lastStock) {
+        return options.showDecrease;
+    } else if (lastStock != 0) {
+        return options.showIncrease;
+    } else {
+        return options.showIncreaseFromZero;
+    }
+}
+
+function limitItems(businessesByItem) {
+    return Object.keys(businessesByItem)
+        .filter(key => {
+                const current = businessesByItem[key];
+                const last = cache.businessesByItem[key];
+                return hasInterestingChange(current, last);
+        })
+        .map(key => businessesByItem[key]);
+}
+
 function notifyIfChanged(businesses) {
-    const message = createMessage(businesses);
+
+    const businessesByItem = businessesToMap(businesses);
```

Comment:
Rename to `businessesById`. Use [_.keyBy](https://lodash.com/docs/4.17.15#keyBy) instead of own function:
```js
const businessesById = _.keyBy(businesses, 'item.item_id')
```

Reply:
I knew there had to be a library function for this somewhere... Thanks!